packages/core/src/App/Components/Elements/Errors/error-component.jsx

```diff
@@ -1,10 +1,28 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import { useHistory } from 'react-router-dom';
 import { PageError } from '@deriv/components';
 import { routes } from '@deriv/shared';
 import { Localize } from '@deriv/translations';
 
-const ErrorComponent = ({ header, message, redirect_label, redirectOnClick, should_show_refresh = true }) => {
+const ErrorComponent = ({
+    header,
+    message,
+    redirect_label,
+    redirectOnClick,
+    should_clear_error_on_click,
+    setError,
+    redirect_to = routes.trade,
+    should_show_refresh = true,
+}) => {
+    const history = useHistory();
+
+    React.useEffect(() => {
+        return history.listen(() => {
```

Comment:
do you want this to happen on unmounting?, if yes then
```
return () => {
 history.listen.......
}
```

Reply:
`history.listen` returns an `unlisten` function which in turn we're returning from **useEffect** so it will be called on unmounting.